### PR TITLE
Doc/dg and ug update

### LINKED ISSUE #110: Update Developer Guide
Update https://github.com/AY2324S1-CS2103T-T10-4/tp/blob/master/docs/DeveloperGuide.md to match the new code we write.

Try to update this frequently so that new features can be implemented quickly.

### LINKED ISSUE #111: Update User Guide
Update the user guide at https://github.com/AY2324S1-CS2103T-T10-4/tp/blob/master/docs/UserGuide.md for the commands that you write.

### PR DESCRIPTION
**Pull request by rngzhi**

### CODE REVIEW

**Comment by Tim-Siu** [comment]
This is a holistic update. Will look through the details later.

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -552,21 +552,99 @@ testers are expected to do more *exploratory* testing.
...
+1. Adding a person
 
-1. Deleting a person while all persons are being shown
+    1. Test case: `add p n/Tom p/12345678 e/tom@gmail.com a/Kent Ridge Road t/friend`<br>
+       Expected: A person contact with the given information is added to the end of the person contact list. 
+            Details of the newly added person contact shown in the status message. Timestamp in the status bar is updated.
 
-   1. Prerequisites: List all persons using the `list` command. Multiple persons in the list.
+    1. Test case: `add p n/Tom`<br>
+       Expected: No person is added due to field missing(Only the tag field with `t/` prefix is optional). Error details shown in the status message. Status bar remains the same.
```
Comment (lunaroddity) [inline docs/DeveloperGuide.md]: Missing space between 'missing' and '(Only...'. Will merge and correct it on my end in the interest of time

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -552,21 +552,99 @@ testers are expected to do more *exploratory* testing.
...
 
-   1. Test case: `delete 0`<br>
+2. Adding a company
+
+    1. Test case: `add c n/ByteDance p/12345678 e/bytedance@example.com d/Cool Company t/social media`<br>
+       Expected: A company contact with the given information is added to the end of the company contact list.
+       Details of the newly added company contact shown in the status message. Timestamp in the status bar is updated.
+
+    1. Test case: `add c n/ByteDance`<br>
+       Expected: No company is added due to field missing(Only the tag field with `t/` prefix is optional). Error details shown in the status message. Status bar remains the same.
```
Comment (lunaroddity) [inline docs/DeveloperGuide.md]: Missing space between 'missing' and '(Only...'. Will merge and correct it on my end in the interest of time

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -552,21 +552,99 @@ testers are expected to do more *exploratory* testing.
...
 
-### Deleting a person
+### Command Testing
+#### Adding A Person or A Company
+1. Adding a person
 
-1. Deleting a person while all persons are being shown
+    1. Test case: `add p n/Tom p/12345678 e/tom@gmail.com a/Kent Ridge Road t/friend`<br>
+       Expected: A person contact with the given information is added to the end of the person contact list. 
+            Details of the newly added person contact shown in the status message. Timestamp in the status bar is updated.
```
Comment (lunaroddity) [inline docs/DeveloperGuide.md:561]: I'm not quite sure what the timestamp in the status bar is. Do you have a screenshot of it?

File: docs/UserGuide.md [hunk 4]
```diff
@@ -135,6 +137,26 @@ Examples:
 * `list c` followed by `view c 1` views the first contact in the companies list:
   ![show the information of the first company](images/viewCompany.png)
 
+### Editing a person's contact: `edit p`
+
+Edit the information of a person's contact
+
+Format: `edit p [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [a/ADDRESS] [t/TAG]`
```
Comment (lunaroddity) [inline docs/UserGuide.md]: Is this missing the `INDEX` parameter?

File: docs/UserGuide.md [hunk 5]
```diff
@@ -135,6 +137,26 @@ Examples:
...
+* At least one of `NAME`, `PHONE_NUMBER`, `EMAIL`, `ADDRESS` and `TAG` fields should be included.
+
+Examples:
+* `edit p n/John p/98765432 e/john@example.com a/311, Clementi Ave 2, #02-26 t/friend`
```
Comment (lunaroddity) [inline docs/UserGuide.md]: Is this missing the `INDEX `parameter?

File: docs/UserGuide.md [hunk 6]
```diff
@@ -135,6 +137,26 @@ Examples:
...
+
+### Editing a company's contact: `edit c`
+
+Edit the information of a company's contact
+
+Format: `edit c [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [d/DESCRIPTION] [t/TAG]`
```
Comment (lunaroddity) [inline docs/UserGuide.md]: Is this missing the `INDEX `parameter?

File: docs/UserGuide.md [hunk 7]
```diff
@@ -135,6 +137,26 @@ Examples:
...
+* At least one of `NAME`, `PHONE_NUMBER`, `EMAIL`, `DESCRIPTION` and `TAG` fields should be included.
+
+Examples:
+* `edit c n/Alpha p/98765432 e/alpha@example.com d/A cool company t/tech`
```
Comment (lunaroddity) [inline docs/UserGuide.md]: Is this missing the `INDEX `parameter?

File: docs/UserGuide.md [hunk 8]
```diff
@@ -203,11 +225,12 @@ _Details coming soon ..._
...
+| Delete | `delete c INDEX`, `delete p INDEX` | e.g., `delete p 3`                                                                                                                                                                                        |
+| List   | `list c`, `list p`     |                                                                                                                                                                                                           |
+| View   | `view c INDEX`, `view p INDEX`   | e.g., `view c 1`                                                                                                                                                                                          |
+| Edit   |`edit c`, `edit p`| e.g., `edit p n/John p/98765432 e/john@example.com a/311, Clementi Ave 2, #02-26 t/friend`;<br> `edit c n/Alpha p/98765432 e/alpha@example.com d/A cool company t/tech`                                       |                                                                                                                                                                                    |
```
Comment (lunaroddity) [inline docs/UserGuide.md]: Are the formats and examples missing the `INDEX `parameter?

**Review by lunaroddity — APPROVED** [review]

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (rngzhi) [reply docs/DeveloperGuide.md]: Fixed in the new commit.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (rngzhi) [reply docs/DeveloperGuide.md]: Also fixed in the new commit.

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (rngzhi) [reply docs/DeveloperGuide.md:561]: I refer to the AB3 UG. It indicates the message shown in the status bar I think.

File: docs/UserGuide.md (hunk 4 above)
Comment (rngzhi) [reply docs/UserGuide.md]: Yes, I forgot the INDEX field. Also fixed.

Comment (rngzhi) [reply docs/UserGuide.md]: Same for the following issues you raised. Thanks for pointing out.

**Pull request merged**